src/IntelliTect.Coalesce/DataAnnotations/HttpMethodAttribute.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace IntelliTect.Coalesce.DataAnnotations
+{
+    /// <summary>
+    /// Allows specifying if this method should use an HTTP method type other than POST.
+    /// </summary>
+    [System.AttributeUsage(System.AttributeTargets.Method)]
+    public class HttpMethodAttribute : System.Attribute
+    {
+        public enum HttpMethodType
```

Comment:
Why are we duplicating an enum already available in .net core? https://source.dot.net/#System.Net.Http/System/Net/Http/HttpMethod.cs

Reply:
Or https://source.dot.net/#System.Text.Http.Parser/HttpMethod.cs because it needs to be a compile-time constant (as @Keboo pointed out)

Reply:
The intent was to not take a dependency on asp.net. Both have options that don't really make sense. I didn't see any use in methods for things other than POST, GET, PUT, and DELETE. After some review I think PATCH also makes sense. Updated naming to be more in sync with above names.